beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -304,7 +304,8 @@ func (s *Service) isProcessedBlock(ctx context.Context, blk block.SignedBeaconBl
 	if err != nil {
 		return false
 	}
-	if blk.Block().Slot() <= finalizedSlot || (s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)) {
+	blockExistsInDB := s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)
+	if blk.Block().Slot() <= finalizedSlot || blockExistsInDB && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```

Comment:
Curious on whether we have unit test for this as well

Reply:
we do have a unit test, which correctly failed when I had an off by 1 :) https://github.com/prysmaticlabs/prysm/pull/9452/commits/09c3beeb34caf3cd50ef5b47ae3c8a1eea3d89da